Grammar/python.gram

```diff
@@ -564,14 +565,15 @@ star_targets_seq[asdl_seq*]: a=','.star_target+ [','] { a }
 star_target[expr_ty] (memo):
     | '*' a=(!'*' star_target) {
         _Py_Starred(CHECK(_PyPegen_set_expr_context(p, a, Store)), Store, EXTRA) }
-    | a=t_primary '.' b=NAME !t_lookahead { _Py_Attribute(a, b->v.Name.id, Store, EXTRA) }
-    | a=t_primary '[' b=slices ']' !t_lookahead { _Py_Subscript(a, b, Store, EXTRA) }
+    | a=t_primary '.' b=NAME &star_target_end { _Py_Attribute(a, b->v.Name.id, Store, EXTRA) }
+    | a=t_primary '[' b=slices ']' &star_target_end { _Py_Subscript(a, b, Store, EXTRA) }
     | star_atom
 star_atom[expr_ty]:
-    | a=NAME { _PyPegen_set_expr_context(p, a, Store) }
+    | a=NAME &star_target_end { _PyPegen_set_expr_context(p, a, Store) }
     | '(' a=star_target ')' { _PyPegen_set_expr_context(p, a, Store) }
     | '(' a=[star_targets_seq] ')' { _Py_Tuple(a, Store, EXTRA) }
     | '[' a=[star_targets_seq] ']' { _Py_List(a, Store, EXTRA) }
+star_target_end: ')' | ']' | ',' | '=' | 'in'
```

Comment:
I presume `in` is here because of `for <target> in <expr>`, but this would mean that you'd get a different error for
```
a in b = 42
```
than you'd get for
```
a < b = 42
```

Reply:
Actually not, because when parsing the invalid target the whole expression gets parsed, which means that `cannot assign to comparison` is displayed for both. The problem with this is that it isn't really usable in for-statements, because the `in` always gets parsed as the comparison operator `in`, which means that even something like `for f() in a: pass` displays the `cannot assign to comparison` error.